### DISCUSSION TITLE
Skeleton loader vs spinner loader

### DISCUSSION THREAD

**Discussion by charleneyjh1** [twilio-labs/paste#2022] [Q&A] [answered]
Hi team, I wonder does DSYS has any recommendations on when to use the skeleton loader vs spinner loader?
From my understanding, skeleton loader is always preferred since it improves the perceived performance of the page. I'd like to confirm it with the team. 

Thanks!

**Comment by gloriliale** [comment, marked as answer]
Hi @charleneyjh1! Our general guidance is that spinners should go inside of already-rendered components, like a Button for example, when an action is being performed, but shouldn't be a stand-in for an entire UI that hasn't rendered yet. In that case, our suggestion is going with Skeleton Loader. 

We also plan to have a Loading Pattern early next year, keep an eye out for that, and feel free to contribute to it as well!

**Reply by charleneyjh1** [reply]
Thanks for you reply, Glorilí! Excited for the loading pattern!
